### DISCUSSION TITLE
vorta configuration and borg logging output missing

### DISCUSSION THREAD

**Discussion by frank-github** [borgbase/vorta#1209] [General]
Hi,

a) where can I find the vorta configuration? 
I want to backup my settings ;-)

b) vorta has it own logs and link to the location, but where can I find the borg output ?

bye,   frank

**Comment by frank-github** [comment]
was this too stupid questions?
Even if the answers should be obvious, I don't know them. 
If someone could show me the way, I would be very grateful 
@m3nu do you have a hint for me?

**Reply by real-yfprojects** [reply]
> was this too stupid questions?

No they weren’t.

I think most borg output is in vorta’s logs.

The settings a located in the file `settings.db` located in the folder provided by `user_data_dir` in the [appdirs package](https://pypi.org/project/appdirs/).

**Reply by frank-github** [reply]
thanks for hint to appdirs, at this place I didn't looked before, but settings.db can be found :-)

No, borg write much more output as can be found in vorta log. Information about borg is quite small :-(

**Reply by cantappa** [reply]
Yes, in particular the output produced by the `--list` parameter is unfortunately not contained in vorta's log.

But independently from that: vorta is a really great project! Thanks a lot for it!

**Reply by frank-github** [reply]
yes, vorta is really excellent, but still needs improvement ;-)

I would be happy if 
- you could make the logging more conversational
- export/import the settings
- the complete Borg Command would be displayed on request. (This is also only displayed in parts and is so unfortunately not usable outside Vorta)

**Reply by m3nu** [reply]
We already have an import/export feature.

And Borg commands are logged in full (Misc > Log).

Am I missing something?

**Reply by cantappa** [reply]
Yes, the borg commands itself are logged in full and contained in vorta's log which is already really useful cause those commands can exactly as they are be used for reproducing the borg call vorta makes outside of vorta.

However, (at least) the output of the borg crate command itself is not contained in there, i.e. the JSON which is produced by borg due to the json parameters vorta internally passes to borg create (`--log-json --json`). That JSON output for instance contains the output produced by the `--list` parameter containing information about which items have been changed/deleted/added.

**Reply by frank-github** [reply]
I admit I'm still a Vorta newbie, maybe I haven't found all the options yet.

a) under Debian Bullseye with Vorta 0.7.5 I can't find any option to im/export. Where should this be found?
b) no, the borg-command is apparently not completely in the log, because if I want to execute it in a shell, the wrong ssh-key is used, i.e. Vorta did not really write all its actions into the log.
Vorta writes out that it uses the VortaSecretKeyStore, but not how it then later parameterizes borg to use right keys and passwords

**Reply by m3nu** [reply]
The password is passed as env var and not logged for security. Not sure on the ssh key. Maybe also env var.

**Reply by cantappa** [reply]
For me it also works with having the environment variable `BORG_PASSCOMMAND` accordingly set (in my case in my `~/.zshrc`).

**Reply by frank-github** [reply]
the answer for ssh, env var or --rsh option
> BORG_RSH
    When set, use this command instead of ssh. This can be used to specify ssh options, such as a custom identity file ssh -i /path/to/private/key. See man ssh for other options. Using the --rsh CMD commandline option overrides the environment variable.

**Comment by cantappa** [comment]
I would also be interested in the output of borg itself, in particular in the output coming from --list.
As far as I got it from @samuel-w's answer in [this discussion](https://github.com/borgbase/vorta/discussions/844) this is currently not supported since those log files could be huge. 
However, in that discussion there was also the request to at least have an option to explicitly enable writing those log files which I would also love to have.
